eZ/Publish/Core/Base/Exceptions/DatabaseException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Base\Exceptions;
+
+use RuntimeException;
+use Throwable;
+
+/**
+ * An exception that is thrown when the database encounters an error.
+ */
+final class DatabaseException extends RuntimeException
+{
+    public const DEFAULT_MESSAGE = 'Database error';
+
+    public static function wrap(
```

Comment:
I cannot see the value of this method

Reply:
In all cases that we have we wrap just previous exception with `DatebaseException` so instead of `throw new DatebaseException(DatebaseException::DEFAULT_MESSAGE, 0, $e)` you can simple use  `throw DatebaseException::wrap($e)`.